quic/api/test/Mocks.h

```diff
@@ -348,6 +354,14 @@ class MockInstrumentationObserver : public InstrumentationObserver {
             &InstrumentationObserver::LostPacket::lostByTimeout,
             testing::Eq(timeoutLoss)));
   }
+
+  static auto getLossPacketNum(PacketNum num) {
+    return testing::Field(
+        &OutstandingPacket::packet,
+        testing::Field(
+            &RegularPacket::header,
+            testing::Property(&PacketHeader::getPacketSequenceNum, num)));
```

Comment:
If it's simple, xan you extend `getLostPacketMatcher` to do this check as well? Right now, it's a bit confusing that we check the packet number for spurious retransmits, but not for lost packets.